src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -384,13 +396,26 @@ static SenderThreadParams* buildSenderParamsCustom
   {
     // Important to use const& for Entity here. Otherwise problems may occur in the object release logic
     const Entity& ngsi = (notification.type == ngsiv2::HttpNotification ? notification.httpInfo.ngsi : notification.mqttInfo.ngsi);
-    if (!setNgsiPayload(ngsi, subscriptionId, en, tenant, xauthToken, attrsFilter, blacklist, metadataFilter, &payload))
+    if (!setNgsiPayload(ngsi, subscriptionId, en, tenant, xauthToken, attrsFilter, blacklist, metadataFilter, &payload, renderFormat))
     {
       // Warning already logged in macroSubstitute()
       return NULL;
     }
-    renderFormat = NGSI_V2_NORMALIZED;
-    mimeType = "application/json";
+    if (renderFormat == NGSI_V2_SIMPLIFIEDNORMALIZED)
+    {
+      renderFormat = NGSI_V2_SIMPLIFIEDNORMALIZED;
+      mimeType = "application/json";
+    }
+    else if (renderFormat == NGSI_V2_SIMPLIFIEDKEYVALUES)
+    {
+      renderFormat = NGSI_V2_SIMPLIFIEDKEYVALUES;
+      mimeType = "application/json";
+    }
+    else
+    {
+      renderFormat = NGSI_V2_NORMALIZED;
+      mimeType = "application/json";
+    }
```

Comment:
This is a bit weird... you when a variable to be equal to a give value... and then you set that variable to the same value :)

Maybe this can be replaced by just:

```suggestion
    mimeType = "application/json";
```

I'd suggest to apply the change and if the ftest pass, then go with it. If the ftest not pass, then we should anayze what's going on.

Reply:
Fixed in https://github.com/telefonicaid/fiware-orion/pull/4298/commits/7bf26136d4ee0b7fa8abd10773fcf7ef1e3e043a